src/celeritas/neutron/interactor/ChipsNeutronElasticInteractor.hh

```diff
@@ -0,0 +1,155 @@
+    using Momentum = units::MevMomentum;
+    //!@}
+
+  public:
+    // Construct from shared and state data
+    inline CELER_FUNCTION
+    ChipsNeutronElasticInteractor(NeutronElasticRef const& shared,
+                                  ParticleTrackView const& particle,
+                                  Real3 const& inc_direction,
+                                  IsotopeView const& target);
+
+    // Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// TYPES ////
+
+    using UniformRealDist = UniformRealDistribution<real_type>;
+
+    //// DATA ////
+
+    // Constant shared data
+    NeutronElasticRef const& shared_;
+    // Incident neutron direction
+    Real3 const& inc_direction_;
+    // Target nucleus
+    IsotopeView const& target_;
+
+    // Values of neutron mass (MevMass) and energy (MevEnergy)
+    real_type neutron_mass_;
+    real_type neutron_energy_;
+    Momentum neutron_p_;
+
+    // Sampler
+    UniformRealDist sample_phi_;
+    detail::MomentumTransferSampler sample_momentum_square_;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data, and a target nucleus.
+ */
+CELER_FUNCTION ChipsNeutronElasticInteractor::ChipsNeutronElasticInteractor(
+    NeutronElasticRef const& shared,
+    ParticleTrackView const& particle,
+    Real3 const& inc_direction,
+    IsotopeView const& target)
+    : shared_(shared)
+    , inc_direction_(inc_direction)
+    , target_(target)
+    , neutron_mass_(value_as<Mass>(shared_.neutron_mass))
+    , neutron_energy_(neutron_mass_ + value_as<Energy>(particle.energy()))
+    , neutron_p_(particle.momentum())
+    , sample_phi_(0, 2 * constants::pi)
+    , sample_momentum_square_(shared_, target_, neutron_p_)
+{
+    CELER_EXPECT(particle.particle_id() == shared_.ids.neutron);
+}
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the final state of the neutron-nucleus elastic scattering.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction ChipsNeutronElasticInteractor::operator()(Engine& rng)
+{
+    // Scattered neutron with respect to the axis of incident direction
+    Interaction result;
+
+    // The momentum magnitude in the c.m. frame
+    real_type target_mass = value_as<Mass>(target_.nuclear_mass());
+
+    real_type cm_p = value_as<units::MevMomentum>(neutron_p_)
+                     / std::sqrt(1 + ipow<2>(neutron_mass_ / target_mass)
+                                 + 2 * neutron_energy_ / target_mass);
+
+    // Sample the scattered direction from the invariant momentum transfer
+    // squared (\f$ -t = Q^{2} \f$) in the c.m. frame
+    real_type cos_theta
+        = 1 - real_type(0.5) * sample_momentum_square_(rng) / ipow<2>(cm_p);
```

Comment:
It looks like this interactor is basically kinematics calculations wrapped around the momentum sampler, so in the future do you think we could use it for different models that have different $`Q^2`$ distributions?

Reply:
Yes.  The interactor should be same even for other projectiles (i.e. x + nucleus for any x).